`linkerd install --ignore-cluster` is panicking

### What is the issue?

The `--ignore-cluster` flag is causing a panic

### How can it be reproduced?

```
$ bin/linkerd install --ignore-cluster
```


### Logs, error output, etc

```
$ bin/linkerd install --ignore-cluster
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1981ddd]

goroutine 1 [running]:
github.com/linkerd/linkerd2/cli/cmd.install({0x219c3e0, 0xc000056118}, 0x0, {0x2182680, 0xc00000e018}, 0xc0008a7d60?, {0xc0003fc580?, 0xc0000d2900?, 0xc0008a7d38?}, 0x0, ...)
        /home/alpeb/linkerd2/cli/cmd/install.go:160 +0x7d
github.com/linkerd/linkerd2/cli/cmd.newCmdInstall.func1(0xc000379b80?, {0xc00035d600?, 0x1?, 0x1?})
        /home/alpeb/linkerd2/cli/cmd/install.go:131 +0x2e8
github.com/spf13/cobra.(*Command).execute(0xc000379b80, {0xc00035d5d0, 0x1, 0x1})
        /home/alpeb/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0x312c8c0)
        /home/alpeb/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/alpeb/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
main.main()
        /home/alpeb/linkerd2/cli/main.go:37 +0x273
```

### output of `linkerd check -o short`

NA

### Environment

Edge-22.4.1

### Possible solution

_No response_

### Additional context

_No response_

### Would you like to work on fixing this bug?

maybe

--- a/cli/cmd/install.go
+++ b/cli/cmd/install.go
@@ -156,27 +156,29 @@ func install(ctx context.Context, k8sAPI *k8s.KubernetesAPI, w io.Writer, values
 		return err
 	}
 
-	// We just want to check if `linkerd-configmap` exists
-	_, err = k8sAPI.CoreV1().ConfigMaps(controlPlaneNamespace).Get(ctx, k8s.ConfigConfigMapName, metav1.GetOptions{})
-	if err == nil {
-		fmt.Fprintf(os.Stderr, errMsgLinkerdConfigResourceConflict, controlPlaneNamespace, "ConfigMap/linkerd-config already exists")
-		os.Exit(1)
-	}
-	if !kerrors.IsNotFound(err) {
-		return err
-	}
-
 	// Create values override
 	valuesOverrides, err := options.MergeValues(nil)
 	if err != nil {
 		return err
 	}
 
-	if !isRunAsRoot(valuesOverrides) {
-		err = healthcheck.CheckNodesHaveNonDockerRuntime(ctx, k8sAPI)
-		if err != nil {
-			fmt.Fprintln(os.Stderr, err)
+	if k8sAPI != nil {
+		// We just want to check if `linkerd-configmap` exists
+		_, err = k8sAPI.CoreV1().ConfigMaps(controlPlaneNamespace).Get(ctx, k8s.ConfigConfigMapName, metav1.GetOptions{})
+		if err == nil {
+			fmt.Fprintf(os.Stderr, errMsgLinkerdConfigResourceConflict, controlPlaneNamespace, "ConfigMap/linkerd-config already exists")
 			os.Exit(1)
+		}
+		if !kerrors.IsNotFound(err) {
+			return err
+		}
+
+		if !isRunAsRoot(valuesOverrides) {
+			err = healthcheck.CheckNodesHaveNonDockerRuntime(ctx, k8sAPI)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(1)
+			}
 		}
 	}
 

--- a/cli/cmd/install_test.go
+++ b/cli/cmd/install_test.go
@@ -244,7 +244,7 @@ func TestRender(t *testing.T) {
 	}
 }
 
-func TestIgnoreClutster(t *testing.T) {
+func TestIgnoreCluster(t *testing.T) {
 	defaultValues, err := testInstallOptions()
 	if err != nil {
 		t.Fatal(err)

--- a/cli/cmd/install_test.go
+++ b/cli/cmd/install_test.go
@@ -244,6 +244,19 @@ func TestRender(t *testing.T) {
 	}
 }
 
+func TestIgnoreClutster(t *testing.T) {
+	defaultValues, err := testInstallOptions()
+	if err != nil {
+		t.Fatal(err)
+	}
+	addFakeTLSSecrets(defaultValues)
+
+	var buf bytes.Buffer
+	if err := install(context.Background(), nil, &buf, defaultValues, nil, false, values.Options{}); err != nil {
+		t.Fatal(err)
+	}
+}
+
 func TestRenderCRDs(t *testing.T) {
 	defaultValues, err := testInstallOptions()
 	if err != nil {

